File: src/main/intellibitz/sted/ui/DocumentListenerTextField.java
package sted.ui;

import sted.fontmap.FontMap;
import sted.fontmap.SampleTextConverter;
import sted.io.Resources;
import sted.widgets.FontChangeTextField;

import javax.swing.*;
import javax.swing.event.DocumentEvent;
import javax.swing.event.DocumentListener;

public class DocumentListenerTextField
        extends FontChangeTextField
        implements DocumentListener {
    private SampleTextConverter converter;
    private FontMap fontMap;
    private MapperPanel mapperPanel;
//    private STEDWindow stedWindow;

    public DocumentListenerTextField() {
        super();
    }

/*
    public DocumentListenerTextField(MapperPanel mapperPanel,
            STEDWindow stedWindow)
    {
        this();
        this.mapperPanel = mapperPanel;
        this.stedWindow = stedWindow;
    }
*/

    public void load() {

    }

    public void insertUpdate(DocumentEvent e) {
        convertSampleText(e);
    }

    public void removeUpdate(DocumentEvent e) {
        convertSampleText(e);
    }

    public void changedUpdate(DocumentEvent e) {
        convertSampleText(e);
    }

    private void convertSampleText(DocumentEvent e) {
        if (e.getDocument().getLength() > 0) {
            if (converter == null) {
                converter = new SampleTextConverter(mapperPanel);
            }
            converter.setFontMap(fontMap);
            final MenuHandler menuHandler = MenuHandler.getInstance();
            final JCheckBoxMenuItem preserve =
                    (JCheckBoxMenuItem) menuHandler.getMenuItem
                            (Resources.ACTION_PRESERVE_TAGS);
            converter.setHTMLAware(preserve.isSelected());
            final JCheckBoxMenuItem reverse =
                    (JCheckBoxMenuItem) menuHandler.getMenuItem
                            (Resources.ACTION_TRANSLITERATE_REVERSE);
            converter.setReverseTransliterate(reverse.isSelected());
            SwingUtilities.invokeLater(converter);
        }
    }

    public void setFontMap(FontMap fontMap) {
        this.fontMap = fontMap;
    }

    public void setFontMapperPanel(MapperPanel mapperPanel) {
        this.mapperPanel = mapperPanel;
    }

}

File: src/main/intellibitz/sted/ui/DocumentListenerTextField.kt
package sted.ui

import sted.fontmap.FontMap
import sted.fontmap.SampleTextConverter
import sted.widgets.FontChangeTextField
import javax.swing.JCheckBoxMenuItem
import javax.swing.SwingUtilities
import javax.swing.event.DocumentEvent
import javax.swing.event.DocumentListener

class DocumentListenerTextField : FontChangeTextField(), DocumentListener {
    lateinit var fontMap: FontMap
    lateinit var mapperPanel: MapperPanel
    val converter: SampleTextConverter by lazy {
        SampleTextConverter()
    }

    /*
    public DocumentListenerTextField(MapperPanel mapperPanel,
            STEDWindow stedWindow)
    {
        this();
        this.mapperPanel = mapperPanel;
        this.stedWindow = stedWindow;
    }
*/
    fun load() {}
    override fun insertUpdate(e: DocumentEvent) {
        convertSampleText(e)
    }

    override fun removeUpdate(e: DocumentEvent) {
        convertSampleText(e)
    }

    override fun changedUpdate(e: DocumentEvent) {
        convertSampleText(e)
    }

    private fun convertSampleText(e: DocumentEvent) {
        if (e.document.length > 0) {
            converter.mapperPanel = mapperPanel
            converter.fontMap = fontMap
            val preserve = MenuHandler.menuItems["Preserve <Tags>"] as JCheckBoxMenuItem?
            converter.setHTMLAware(preserve!!.isSelected)
            val reverse = MenuHandler.menuItems["Reverse Transliterate"] as JCheckBoxMenuItem?
            converter.setReverseTransliterate(reverse!!.isSelected)
            SwingUtilities.invokeLater(converter)
        }
    }

}